client/components/EventsPlanningFilters/EditFilterSchedule.tsx

```diff
@@ -99,30 +111,116 @@ export class EditFilterScheduleComponent extends React.Component<IProps, IState>
         this.setState({schedule});
     }
 
+    updateHour(index: number, value: string) {
+        const hours = [...(this.state.schedule.hours ?? [])];
+
+        hours[index] = value;
+        this.setState({
+            schedule: {...this.state.schedule, hours},
+            pristine: false,
+        });
+    }
+
+    addHour() {
+        const used = new Set(this.state.schedule.hours ?? []);
+        const allHours = Array.from({length: 24}, (_, i) => `${i.toString().padStart(2, '0')}:00`);
+        const next = allHours.find((h) => !used.has(h)) || '00:00';
+
+        const hours = [...(this.state.schedule.hours ?? []), next];
+
+        this.setState({
+            schedule: {...this.state.schedule, hours},
+            pristine: false,
+        });
+    }
+
+    removeHour(index: number) {
+        const hours = [...(this.state.schedule.hours ?? [])];
+
+        hours.splice(index, 1);
+        this.setState({
+            schedule: {...this.state.schedule, hours},
+            pristine: false,
+        });
+    }
+
     getScheduleProfile() {
         const profile = {
             frequency: {enabled: true, index: 1},
             week_days: {enabled: false, index: 2},
             month_day: {enabled: false, index: 3},
-            hour: {enabled: false, index: 4},
         };
 
         switch (this.state.schedule.frequency) {
-        case SCHEDULE_FREQUENCY.HOURLY:
-            break;
         case SCHEDULE_FREQUENCY.WEEKLY:
             profile.week_days.enabled = true;
-            profile.hour.enabled = true;
             break;
         case SCHEDULE_FREQUENCY.MONTHLY:
             profile.month_day.enabled = true;
-            profile.hour.enabled = true;
             break;
         }
 
         return profile;
     }
 
+    renderTimeInputs() {
```

Comment:
Do not return JSX from class methods. Either inline it if it's the only usage or extract into a separate component.